pkg/querier/worker/worker.go

```diff
@@ -21,22 +22,25 @@ import (
 	"github.com/weaveworks/common/httpgrpc"
 	"google.golang.org/grpc"
 
+	"github.com/grafana/mimir/pkg/scheduler/schedulerdiscovery"
 	"github.com/grafana/mimir/pkg/util"
 )
 
 type Config struct {
-	FrontendAddress       string        `yaml:"frontend_address"`
-	SchedulerAddress      string        `yaml:"scheduler_address"`
-	DNSLookupPeriod       time.Duration `yaml:"dns_lookup_duration" category:"advanced"`
-	MaxConcurrentRequests int           `yaml:"-"` // Must be same as passed to PromQL Engine.
-	QuerierID             string        `yaml:"id" category:"advanced"`
-
+	FrontendAddress  string            `yaml:"frontend_address"`
+	SchedulerAddress string            `yaml:"scheduler_address"`
+	DNSLookupPeriod  time.Duration     `yaml:"dns_lookup_duration" category:"advanced"`
+	QuerierID        string            `yaml:"id" category:"advanced"`
 	GRPCClientConfig grpcclient.Config `yaml:"grpc_client_config"`
+
+	// This configuration is injected internally.
+	MaxConcurrentRequests   int                       `yaml:"-"` // Must be same as passed to PromQL Engine.
+	QuerySchedulerDiscovery schedulerdiscovery.Config `yaml:"-"`
 }
 
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
-	f.StringVar(&cfg.SchedulerAddress, "querier.scheduler-address", "", "Address of the query-scheduler component, in host:port format. Only one of -querier.frontend-address or -querier.scheduler-address can be set. If neither is set, queries are only received via HTTP endpoint.")
-	f.StringVar(&cfg.FrontendAddress, "querier.frontend-address", "", "Address of the query-frontend component, in host:port format. Only one of -querier.frontend-address or -querier.scheduler-address can be set. If neither is set, queries are only received via HTTP endpoint.")
+	f.StringVar(&cfg.SchedulerAddress, "querier.scheduler-address", "", fmt.Sprintf("Address of the query-scheduler component, in host:port format. If multiple query-schedulers are running, the host should be a DNS resolving to all query-scheduler instances. This option should be set only when query-scheduler component is in use and -%s is set to '%s'.", schedulerdiscovery.ModeFlagName, schedulerdiscovery.ModeDNS))
```

Comment:
"the host should be a DNS resolving to all query-scheduler instances".

We can either say "the host should resolve ..." or "the host should be a DNS *entry* that resolves ...". I prefer shorter version.